agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/sql/PreparedStatementClassDataProvider.java

```diff
@@ -59,8 +59,10 @@ public void add() {
             factory = classFactoryForMySql();
             doAdd(factory, "com/mysql/jdbc/PreparedStatement");
 
-            factory = classFactoryForPostgreSql();
+            factory = classFactoryForOldPostgreSql();
             doAdd(factory, "org/postgresql/jdbc2/AbstractJdbc2Statement");
+            factory = classFactoryForNewPostgreSql();
```

Comment:
Probably a better renaming here is recommended. New and Old doesn't seem to be old. Recommend being specific about the version of Postgres being instrumented.

Reply:
Ok, updated to jdbc2 and jdbc4

Reply:
This look much better :)